jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -47,6 +47,7 @@
     private final PropagationRegistry registry;
     private final String serviceName;
     private final Metrics metrics;
+    private SpanListener spanListener;
```

Comment:
* Tracer should contain an EventListenerRegistry instance, always non-empty
* registry itself can implement EventListener so that tracer can always call `this.eventListenerRegistry.spanReported()` etc.
* tracer should have addEventListener method that will delegate to registry.addEventListener
* registry implements all listener methods by delegating to an internal list of listeners one at a time